### ISSUE TITLE
Use actions based authorization instead of request method

### ISSUE THREAD

**Issue by olhybrius** [marcelo225/django-keycloak-auth#4]
Hi,

Using actions instead of request methods to authorize resources would allow support for custom actions and more fined-grained control (multiple actions can use the same request method).

Updating line 87 of middleware.py like below would do the trick :

```
require_view_role = view_roles.get(self.action, [None]) 
```

I can make a PR if you want, and update the README accordingly.

**Comment by marcelo225** [comment]
@olhybrius 
good ideia. I even didn't know about it. Hahah. Could you help me to create a PR before i approve it?
Don't forget to create unit tests about new functionalities. Right?
Thanks!

**Comment by olhybrius** [comment]
@marcelo225 
Sure, will do when I find time, probably some time during the week.

**Comment by marcelo225** [comment]
@olhybrius 
I'll analyze your PR as soon as possible during the week. It'll help several people. 
Thanks for contributing to it.

**Comment by olhybrius** [comment]
I did some research last night. Turns out it won't be as easy as I thought, if it's doable at all.
The thing is, the documentation clearly states that the view_func parameter passed to the process_view function of a Django middleware is not the actual view instance, but the function that Django is about to use. That means there is no way to access the instance variables of the view, and the current action is one of them.
However, we do have access to all the available actions on the view since they're passed as argument to the as_view function of a viewset, as well as the user-defined extra actions since they're returned by the get_extra_actions class method, so there may be some way around. But we still need to find a way to determine the current action somehow.
